src/internal/components/pluggableVisualizations/pivotTable/sortItemsHelpers.ts

```diff
@@ -0,0 +1,191 @@
+// (C) 2020 GoodData Corporation
+import includes = require("lodash/includes");
+import flatMap = require("lodash/flatMap");
+import { AFM, VisualizationObject } from "@gooddata/typings";
+import * as BucketNames from "../../../../constants/bucketNames";
+import {
+    IBucketItem,
+    IBucketFilter,
+    isAttributeFilter,
+    IAttributeFilter,
+} from "../../../interfaces/Visualization";
+// removes attribute sortItems with invalid identifiers
+// removes measure sortItems with invalid identifiers and invalid number of locators
+function adaptSortItemsToPivotTable(
+    originalSortItems: AFM.SortItem[],
+    measureLocalIdentifiers: string[],
+    rowAttributeLocalIdentifiers: string[],
+    columnAttributeLocalIdentifiers: string[],
+): AFM.SortItem[] {
+    const attributeLocalIdentifiers = [...rowAttributeLocalIdentifiers, ...columnAttributeLocalIdentifiers];
+
+    return originalSortItems.reduce((sortItems: AFM.SortItem[], sortItem: AFM.SortItem) => {
```

Comment:
could this be somehow improved it's really unreadable; maybe with simple for loop?

Reply:
Done. Check it please: https://github.com/gooddata/gooddata-react-components/pull/1705/commits/1f845a1e401ad358a0c12dd3dfc00e52ff84fecb